### DISCUSSION TITLE
Same client ID with app and web applications?

### DISCUSSION THREAD

**Discussion by m1212e** [zitadel/zitadel#8253] [Q&A] [answered]
Hi,

I'm, trying to host kitchenowl on my server (https://docs.kitchenowl.org/self-hosting/oidc/) which requires a (ONE) client ID to be configured. This client must allow the redirect URIs for the web and the app redirect:
![image](https://github.com/zitadel/zitadel/assets/14091540/233cd587-a262-414e-9248-fdad92842ae5)
But when I add a second, native app which would allow the redirect scheme, it has a different client ID. Is this even possible to achieve with ZITADEL?

Thanks!

**Comment by hifabienne** [comment]
Hi @m1212e 
We generally recommend to have one app per client as they have different use cases.
With the app creation flow we configure the app directly for the choosen type.
But you can change to advanced configuration there you can configure what every you like.
I think with that it is possible to configure the app to enable your web and native apps

**Reply by m1212e** [reply]
Hey, thanks for your response. I cannot really find what you were referencing in your response. Where in the admin UI of ZITADEL can I find those config you were talking about?

**Reply by hifabienne** [reply]
If you are in the creation steps of an app, you can click I am a pro Skip this wizard, and you are free to configure what you want.
<img width="1335" alt="image" src="https://github.com/zitadel/zitadel/assets/38692350/29d1257b-8c2c-49b0-9f45-0f7906d15aa6">
With that you can allow all grant types, response types, etc.
<img width="1335" alt="image" src="https://github.com/zitadel/zitadel/assets/38692350/9b2f1665-8b81-4018-aeb1-6f7ea01ee993">

Also on the detail page of the app, you can change the configuration to what ever you like
<img width="1335" alt="image" src="https://github.com/zitadel/zitadel/assets/38692350/67a63353-6c7c-440b-bf8c-6f8b74fde81e">

**Reply by m1212e** [reply]
![image](https://github.com/zitadel/zitadel/assets/14091540/334dd4fb-8d27-4daa-9fca-96bab8310cc3)
I see. But when I now try to add the app redirect URL it still tells me this is not possible.

**Reply by hifabienne** [reply, marked as answer]
I think it is just a warning, as this is not a compliant oidc configuration, but you can still use it.
If not try to activate developement mode

**Comment by tomcurran2** [comment]
At **[Hosting Mastery Hub](https://hostingmasteryhub.com/)**, we understand the intricacies involved in hosting applications like KitchenOwl. From the information provided, it seems like you're encountering an issue with configuring a single client ID that can accommodate multiple redirect URIs for both the web and native app.

ZITADEL, as an Identity and Access Management (IAM) solution, should allow the configuration of multiple redirect URIs under a single client ID. However, it’s crucial to ensure that the redirect URIs are properly registered in the client settings.

Here's a step-by-step approach you can try:

Single Client ID Configuration:

In ZITADEL, navigate to the application/client settings.
Ensure that both the web and native app redirect URIs are listed under the same client ID. ZITADEL typically supports multiple redirect URIs for a single client.
OIDC (OpenID Connect) Settings:

Double-check the OIDC settings to confirm that they allow multiple URIs.
Ensure that the URIs are compliant with ZITADEL’s requirements (e.g., correct syntax, valid domains).
Native App Redirect Scheme:

If the native app requires a custom scheme (e.g., myapp://redirect), this should be added to the same client configuration under redirect URIs.
ZITADEL Documentation:

Refer to the ZITADEL documentation for specific instructions on handling multiple redirect URIs under a single client. There may be examples or best practices outlined.
If after these steps you’re still facing difficulties, it might be beneficial to reach out to ZITADEL support for more tailored assistance. Hosting Mastery Hub always recommends consulting the official documentation and support channels of the tools you are using to ensure the best configuration.

We hope this helps you in hosting KitchenOwl smoothly on your server. If you have any further questions or need more assistance, feel free to ask!

**Reply by m1212e** [reply]
This sounds kinda AI generated to be honest. But thanks I guess.
